openmmtools/tests/test_storage_iodrivers.py

```diff
@@ -0,0 +1,209 @@
+            del data_append, data_append_out
+        # Reopen and test reading actions
+        nc_io_driver = NetCDFIODriver(file_path, access_mode='r')
+        data_write = nc_io_driver.get_storage_variable(write_path)
+        if with_append:
+            data_append = nc_io_driver.get_storage_variable(append_path)
+        # Test unbound read
+        data_write_out = data_write.read()
+        if with_append:
+            data_append_out = data_append.read()
+        try:  # Compound dictionary processing
+            for key in data_write_out.keys():
+                assert np.all(data_write_out[key] == input_data[key])
+        except AttributeError:
+            assert np.all(data_write_out == input_data)
+        if with_append:
+            try:
+                for key in data_write_out.keys():
+                    assert np.all(data_append_out[0][key] == input_data[key])
+                    assert np.all(data_append_out[1][key] == input_data[key])
+            except AttributeError:
+                assert np.all(data_append_out[0] == input_data)
+                assert np.all(data_append_out[1] == input_data)
+
+
+def test_netcdf_int_type_codec():
+    """Test that the Int type codec can read/write/append"""
+    input_data = 4
+    generic_type_codec_check(input_data)
+
+
+def test_netcdf_float_type_codec():
+    """Test that the Float type codec can read/write/append"""
+    input_data = 4.0
+    generic_type_codec_check(input_data)
+
+
+def test_netcdf_string_type_codec():
+    """Test that the String type codec can read/write/append"""
+    input_data = 'four point oh'
+    generic_type_codec_check(input_data)
+
+
+def test_netcdf_list_type_codec():
+    """Test that the List type codec can read/write/append"""
+    input_data = [4, 4, 4]
+    generic_type_codec_check(input_data)
+
+
+def test_netcdf_tuple_type_codec():
+    """Test that the tuple type codec can read/write/append"""
+    input_data = (4, 4, 4)
+    generic_type_codec_check(input_data)
+
+
+def test_netcdf_array_type_codec():
+    """Test that the ndarray type codec can read/write/append"""
+    input_data = np.array([4, 4.0, 4])
+    generic_type_codec_check(input_data)
+
+
+def test_netcdf_quantity_type_codec():
+    """Test that the simtk.unit.Quantity type codec can read/write/append with various unit and _value types"""
+    input_data = 4 * unit.kelvin
+    generic_type_codec_check(input_data)
+    input_data = [4, 4, 4] * unit.kilojoules_per_mole
+    generic_type_codec_check(input_data)
+    input_data = np.array([4, 4, 4]) / unit.nanosecond
+    generic_type_codec_check(input_data)
+
+
+def test_netcdf_dictionary_type_codec():
+    """Test that the dictionary type codec can read/write/append with various unit and _value types"""
+    input_data = {
+        'count': 4,
+        'ratio': 0.4,
+        'name': 'four',
+        'repeated': [4, 4, 4],
+        'temperature': 4 * unit.kelvin,
+        'box_vectors': (np.eye(3) * 4.0) * unit.nanometer
```

Comment:
Can you add a `Quantity`-wrapped numpy array as a test case here to make sure it works if it's not somewhere else?